Fix Apollo cache

nikomakela opened the pull request:
LIIKUNTA-439 HH-297.

Critical fix to the Apollo-client cache that didn't persist.

Steps to test: 

1. Make a search (in Events / Hobbies / Sports, venues or events)
2. Click a result card to navigate to the details page
3. Click back arrow or browser back button
→ The browser should not make a new query but scroll to the card that was clicked earlier, but instead it makes a new query

----

LIIKUNTA-432

Fixed the locations selected module in the article page. 
Needed also to update the CMS content related to the languages (Wordpress PolyLang):
1. https://liikunta.hkih.stage.geniem.io/wp-admin/admin.php?page=mlang
2. https://liikunta2.content.api.hel.fi/wp-admin/admin.php?page=mlang

nikomakela committed:
fix: apollo cache

LIIKUNTA-439 HH-297.

Critical fix to the Apollo-client cache that didn't persist.

nikomakela commented:
The error boundary should be tested!

- [ ] Test that the error boundary is still working. E.g use local router and shut it down in middle of the browsing.
- [ ] Investigate why the `args` in dependency array did not work. Does the args change too often?

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr273.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr273.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr273.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela commented:
It seems like there are some issues with the ErrorBoundary still, but they have been there also before the changes done in this pull request 
<img width="1005" alt="image" src="https://user-images.githubusercontent.com/389204/226742223-7de54657-ce4b-47db-9f9a-cf87953efb07.png">

nikomakela committed:
fix: the locales of the hard coded list of languages

LIIKUNTA-432.

Fixed the locations selected module in the article page.
Needed also to update the CMS content related to the languages (Wordpress PolyLang):
1. https://liikunta.hkih.stage.geniem.io/wp-admin/admin.php?page=mlang
2. https://liikunta2.content.api.hel.fi/wp-admin/admin.php?page=mlang

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr273.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr273.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**EVENTS GRAPHQL PROXY is deployed to platta: https://events-graphql-proxy-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr273.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE GRAPHQL PROXY is deployed to platta: https://venue-graphql-proxy-pr273.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr273.dev.hel.ninja :crying_cat_face::anger::boom::boom:
